lookout/style/typos/benchmarks/typo_commits_report.py

```diff
@@ -0,0 +1,193 @@
+
+    inspected_analyzer_type = IdTyposAnalyzer
+
+    @classmethod
+    def get_report_names(cls) -> Tuple[str, ...]:
+        """
+        Get all available report names.
+
+        :return: Tuple with report names.
+        """
+        return ("report", )
+
+    def _generate_reports(self, dataset_row: Dict[str, Any], fixes: Sequence[TypoFix],
+                          ) -> Dict[str, str]:
+        """
+        Generate reports for the dataset row.
+
+        :param dataset_row: Dataset row which triggered the analyze method of the analyzer.
+        :param fixes: List of `TypoFix`-es provided by the `TyposAnalyzerSpy.analyze()` method.
+        :return: Dictionary with report names as keys and report string as values.
+        """
+        return {
+            "report": self.generate_commit_dataset_report(dataset_row, fixes),
+        }
+
+    def generate_commit_dataset_report(self, dataset_row: Dict[str, Any],
+                                       fixes: Sequence[TypoFix]) -> str:
+        """
+        Generate commit dataset report for the dataset row.
+
+        :param dataset_row: Dataset row which triggered the analyze method of the analyzer.
+        :param fixes: List of `TypoFix`-es provided by the `TyposAnalyzerSpy.analyze()` method.
+        :return: Dictionary with report names as keys and report string as values.
+        """
+        metrics = self.get_metrics_billet()
+
+        correct_fix = dataset_row["correct id"]
+        wrong_identifier = dataset_row["wrong id"]
+        for fix in fixes:
+            if fix.identifier == wrong_identifier:
+                if correct_fix in fix.identifier_candidates:
+                    metrics.detection_true_positive += 1
+                    metrics.fix_accuracy += 1
+            else:
+                metrics.detection_false_positive += 1
+        return json.dumps(metrics.to_dict())
+
+    def _trigger_review_event(self, dataset_row: Dict[str, Any]) -> Sequence[TypoFix]:
+        comments = self._analyzer_context_manager.review(
+            dataset_row["commit"], "HEAD", git_dir=dataset_row["repo"], bblfsh=self._bblfsh,
+            log_level="info", config_json=self._config)
+        typo_fixes = []
+        for comment in comments:
+            typo_fix_dict = json.loads(comment.text)
+            typo_fixes.append(TypoFix(head_file=typo_fix_dict["head_file"],
+                                      line_number=typo_fix_dict["line_number"],
+                                      candidates=typo_fix_dict["candidates"],
+                                      token=typo_fix_dict["token"]))
+        return typo_fixes
+
+    def _finalize(self, reports: Iterable[Dict[str, str]]) -> Iterator[Dict[str, str]]:
+        """
+        Summarize all individual reports to the final one.
+
+        :param reports: Reports generated by `TypoCommitsReporter.generate_commit_dataset_report()`
+        :return: Summarized final report
+        """
+        # TODO(zurk): Add a proper report
```

Comment:
wdym "proper"?

Reply:
Right now it is just some numbers I found useful, no additional information, no details, no jinja template. 
I (or somebody else) should address it later in the next PR.

Reply:
OK, please this TODO list in the comment